### PR TITLE
Support multiple reward models

### PR DESCRIPTION
**Pull request by anfernee**
This is to implement one of the TODO items in the code base from @tsbao. I haven't checked with him this is right approach. Please let me know if it matches what's in his mind or not, I am happy to make changes. 

**Reference**

**Colab Notebook**

**Checklist**

- [x] I have added all the necessary unit tests for my change.
- [x] I have verified that my change does not break existing code and all unit tests pass.
- [x] I have added all appropriate doc-strings/documentation.
- [x] My PR is based on the latest changes of the main branch (if unsure, rebase the code).
- [ ] I have signed the [Contributor License Agreement](https://cla.developers.google.com/about).
- [x] I have followed [Contribution Guidelines](https://github.com/google/tunix/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: tunix/rl/inference/inference_worker.py [hunk 1]
```diff
@@ -25,28 +25,66 @@ class InferenceWorker:
...
       eos_id: int,
+      reward_model_name: str = "reward",
   ) -> jax.Array:
-    reward_model = self._models.get("reward")
+    reward_model = self._models.get(reward_model_name)
     if reward_model is None:
-      raise ValueError("Reward model is not available.")
+      # Try to find any reward model if the specified one doesn't exist
+      reward_models = {k: v for k, v in self._models.items() if k.startswith("reward")}
+      if not reward_models:
```
Comment (wang2yn84) [inline tunix/rl/inference/inference_worker.py:47]: assert reward_models, ... ?

File: tunix/rl/inference/inference_worker.py [hunk 2]
```diff
@@ -25,28 +25,66 @@ class InferenceWorker:
...
+        raise ValueError("No reward model is available.")
+      if len(reward_models) == 1:
+        reward_model = next(iter(reward_models.values()))
+      else:
+        available_models = ", ".join(reward_models.keys())
+        raise ValueError(
+            f"Reward model '{reward_model_name}' is not available. "
```
Comment (wang2yn84) [inline tunix/rl/inference/inference_worker.py:54]: Let's be more specific here, since in line#50 we accepted the only available reward model with prefix matching, let's say something like "f"Reward model '{reward_model_name}' is not available. There are more than 1 reward models with name starting with \'reward\'. Please pick one from available models: {available_models}"

File: tunix/rl/inference/inference_worker.py [hunk 3]
```diff
@@ -25,28 +25,66 @@ class InferenceWorker:
...
+      raise ValueError("No reward models are available.")
+    
+    rewards = {}
+    for model_name, model in reward_models.items():
+      rewards[model_name] = common.compute_score(
+          model, prompt_tokens, completion_tokens, pad_id, eos_id
+      )
+    return rewards
+
+  def get_available_reward_models(self) -> list[str]:
```
Comment (wang2yn84) [inline tunix/rl/inference/inference_worker.py:84]: s/get_available_reward_model/get_available_reward_model_names/?

File: tunix/rl/rl_cluster.py [hunk 4]
```diff
@@ -767,11 +767,33 @@ def get_rewards(
       completion_tokens: jax.Array,
       pad_id: int,
       eos_id: int,
+      reward_model_name: str = "reward",
```
Comment (wang2yn84) [inline tunix/rl/rl_cluster.py:770]: We should stick to this API and make reward_model_name conditional, if it's set, we retrieve rewards based on the provided name, otherwise defaults to retrieve all the rewards. We don't need get_all_rewards().

File: tunix/rl/rl_cluster.py [hunk 5]
```diff
@@ -767,11 +767,33 @@ def get_rewards(
...
+    """Get rewards from all available reward models."""
+    with self.cluster_config.role_to_mesh[Role.REWARD]:
+      return self.inference_worker.get_all_rewards(
+          prompt_tokens,
+          completion_tokens,
+          pad_id,
+          eos_id,
+      )
+
+  def get_available_reward_models(self) -> list[str]:
```
Comment (wang2yn84) [inline tunix/rl/rl_cluster.py:797]: Any reason to add this API?

**Review by wang2yn84 — COMMENTED** [review]
Thank you for your PR. Left some comments.
